### PR TITLE
Activation Notice: Externalize dependencies

### LINKED ISSUE #8576: Activation Notice: Externalize dependencies
## Feature Description

The `activation-notice` package doesn't have any dependencies on `@web-stories-wp/*` besides `i18n` and `tracking`. That first dependency I'd like to remove as there is no need for that wrapper.

Also, React is not loaded as an external dependencies, which means it's part of the bundle, which makes it unnecessarily large.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

First, remove the following code:

https://github.com/google/web-stories-wp/blob/9832e0c60c1ad68740c881c8e9f2bdcf61488aa7/webpack.config.cjs#L328-L330

Then, in the package, use `createInterpolateElement` from `@wordpress/element` instead of `TranslateWithMarkup`.

Update dependencies in `package.json` accordingly.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Removes dependency on `@web-stories-wp/i18n` and ensures that the externally loaded React version provided by WordPress is being used.

## Relevant Technical Choices

Uses `display: block` for links in the activation notice to ensure they're on a new line.

This is needed because `createInterpolateElement` does not support `<br>` tags and thus these cannot be used to create line breaks in these messages.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8576

### CODE REVIEW

File: packages/activation-notice/src/app/components/link.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const Link = styled.a`
   color: ${({ theme }) => theme.colors.link.fg};
   text-decoration: none;
   cursor: pointer;
-
+  display: block;
```
Comment (swissspidy) [inline packages/activation-notice/src/app/components/link.js:30]: Curious, why was this needed? Does this cause any visual changes?

Comment (spacedmonkey) [reply packages/activation-notice/src/app/components/link.js:30]: I removed the br in thing strings. Display block means it goes to a new line

**Comment by spacedmonkey** [comment]
@swissspidy This only saves `-402 B`

File: packages/activation-notice/src/app/components/step1.js [hunk 2]
```diff
@@ -65,6 +64,20 @@ function Step1() {
     trackClick(evt, 'open_demo_story');
   }, []);
 
+  const translatedString = createInterpolateElement(
+    __('Read the<a>Get Started story</a>', 'web-stories'),
```
Comment (swissspidy) [inline packages/activation-notice/src/app/components/step1.js]:
```suggestion
    __('Read the <a>Get Started story</a>', 'web-stories'),
```

Without the `br` I suppose we should add a space everywhere instead.

File: packages/activation-notice/src/app/components/link.js (hunk 1 above)
Comment (swissspidy) [reply packages/activation-notice/src/app/components/link.js:30]: So I take it that `createInterpolateElement` doesn't support `br` then? Would be good to note that in the PR description.

File: packages/activation-notice/src/app/components/step2.js [hunk 3]
```diff
@@ -18,14 +18,13 @@
...
 import styled from 'styled-components';
-import { TranslateWithMarkup } from '@web-stories-wp/i18n';
 import { trackClick } from '@web-stories-wp/tracking';
 
 /**
  * WordPress dependencies
  */
 import { __, _x } from '@wordpress/i18n';
-import { useCallback } from '@wordpress/element';
+import { createInterpolateElement, useCallback } from '@wordpress/element';
```
Comment (swissspidy) [inline packages/activation-notice/src/app/components/step2.js:27]: I think this needs to be mocked in the JS unit tests. They are currently failing.

File: packages/activation-notice/src/app/components/test/step2.js [hunk 4]
```diff
@@ -27,6 +27,11 @@ import { renderWithTheme } from '../../../testUtils';
 import Step2 from '../step2';
 import { ConfigProvider } from '../../config';
 
+jest.mock('@wordpress/element', () => ({
+  ...jest.requireActual('@wordpress/element'),
+  createInterpolateElement: (string) => string,
```
Comment (swissspidy) [inline packages/activation-notice/src/app/components/test/step2.js]: To avoid repetition, I think we can just put this in `__mocks__/@wordpress/element.js`

**Review by swissspidy — APPROVED** [review]
